### DISCUSSION TITLE
Should we add more to the CODEOWNERS file?

### DISCUSSION THREAD

**Discussion by arschles** [gomods/athens#1939] [Ideas]
The [current CODEOWNERS file](https://github.com/gomods/athens/blob/ccf59405cf5c68a55de6c851b225d541f5bf8d50/.github/CODEOWNERS) has only one line - everything is "owned" by @gomods/maintainers. We have a bigger community now, and a super different codebase, than we did when we committed that file.

Should we add more lines to the file so that we have different directories specialized to particular people/teams?

**Comment by michalpristas** [comment]
i would go with teams, which seems easier to manage. dont need to have PR with changes if you want to add somebody.
was thinking about package domain teams:
- storage
- api (incl middlewares)
- communication flow
- security 
- test
- ops (tracing, config)

just from top of my head

**Comment by arschles** [comment]
@michalpristas would those be represented by wildcards in the codeowners file? Also, would we create Github teams for each of those categories?

**Comment by ghost** [comment]
I think we have to create teams in GitHub to assign them to portions of the codebase. I don't believe you can assign more than one person or entity as an owner, one `@something` etc.

**Comment by michalpristas** [comment]
@arschlesI meant in a way how @robjloranger described

**Comment by ghost** [comment]
I like the initial ideas @michalpristas had here https://github.com/gomods/athens/issues/806#issuecomment-432524651

We can always adjust as we move along. I'm happy to create the teams today if that's somewhere we'd like to start.
